We seem to historically prefer pthreads instead of native threads.

https://github.com/lurcher/unixODBC/commit/7d72255824cb96599e71d19fbbc94405db658274#r108478182

Historically we were suggesting -lsocket -lnsl -lgen -z ignore for our users.

--- unixODBC-2.3.13/DriverManager/odbc.pc.in.orig
+++ unixODBC-2.3.13/DriverManager/odbc.pc.in
@@ -17,4 +17,4 @@
 Version: @PACKAGE_VERSION@
 Cflags: -I${includedir}
 Libs: -L${libdir} -lodbc
-Libs.private: @LIBLTDL@ @LIBICONV@ @LIBS@
+Libs.private: @LIBLTDL@ @LIBICONV@ @LIBS@ -lsocket -lnsl -lgen -z ignore
--- unixODBC-2.3.13/configure.ac.orig
+++ unixODBC-2.3.13/configure.ac
@@ -438,9 +438,9 @@
   else
     gotthread="no";
 
-	AC_MSG_CHECKING( if os is AIX )
+	AC_MSG_CHECKING( if $host_os os is AIX or Solaris )
 	case $host_os in
-    	"aix"*)
+    	"aix"|"solaris"*)
 			raw_threads="no";
 			AC_MSG_RESULT( yes - disable check for libthread );
     	;;
--- unixODBC-2.3.13/odbcinst/odbcinst.pc.in.orig
+++ unixODBC-2.3.13/odbcinst/odbcinst.pc.in
@@ -9,4 +9,4 @@
 Version: @PACKAGE_VERSION@
 Cflags: -I${includedir}
 Libs: -L${libdir} -lodbcinst
-Libs.private: @LIBLTDL@ @LIBS@
+Libs.private: @LIBLTDL@ @LIBS@ -lsocket -lnsl -lgen -z ignore
